src/bubble_swarm/utils.rs

```diff
@@ -0,0 +1,75 @@
+use crate::math2d::{circle::Circle, point::Point, regioned_angle::RegionedAngle, vector::Vector};
+
+use super::{angle_area::AngleArea, source_circle::SourceCircle};
+
+pub fn get_mid_point(from: Point, to: Point) -> Point {
+    from + Vector::from((from, to)).scale(0.5)
+}
+
+pub enum CircleIntersection {
+    None,
+    One(Point),
+    Two(Point, Point),
+}
+
+pub fn calculate_circle_intersection(first: &Circle, second: &Circle) -> CircleIntersection {
+    // Calculate the distance between the centers.
+    let distance = Vector::from((first.get_center(), second.get_center())).magnitude();
+
+    // Check if the circles are completely separate.
+    if distance > first.get_radius() + second.get_radius() {
+        return CircleIntersection::None;
+    }
+
+    // Check if one circle is contained within the other.
+    if distance < (first.get_radius() - second.get_radius()).abs() {
+        return CircleIntersection::None;
+    }
+
+    // Check if the circles are touching.
+    if distance == first.get_radius() + second.get_radius() {
+        return CircleIntersection::One(get_mid_point(first.get_center(), second.get_center()));
```

Comment:
get_mid_point is not guaranteed to be correct here. It would only work if the two radiuses are equal.
Leave it as that for now and write comment todo fix me. Cause it needs a bit more thinking how to properly implement this. (The question is whether we introduce a class for it called norm_vector which is guaranteed to be of magnitude 1)